src/main/java/com/google/api/gax/core/NanoClock.java

```diff
@@ -29,23 +29,33 @@
  */
 package com.google.api.gax.core;
 
-/**
- * An interface for getting the current value of a high-resolution time source, in nanoseconds.
- *
- * Clocks other than DefaultNanoClock are typically used only for testing.
- *
- * This interface is required in addition to Java 8's Clock, because nanoTime is required to compare
- * values with io.grpc.CallOptions.getDeadlineNanoTime().
- */
-public interface NanoClock {
+import java.io.ObjectStreamException;
+import java.io.Serializable;
+import java.util.concurrent.TimeUnit;
+
+/** Default implementation of the ApiClock interface, using call to System.nanoTime(). */
+public final class NanoClock implements ApiClock, Serializable {
```

Comment:
The other clock is called SystemClock to indicate that it based on system time (actual time, which is meaningful on any other system). In case of nano clock it can be used only to measure time difference between two events on same machine and only. I.e most important difference between two clocks is NOT the precision, but the actual meaning of the returned value.